e2e/util/namespace.go

```diff
@@ -32,56 +41,86 @@ func CreateNamespace(ctx types.Context, cluster types.Cluster, namespace string)
 			return err
 		}
 
-		log.Debugf("Namespace %q already exist in cluster %q", namespace, cluster.Name)
+		log.Debugf("Namespace %q already exists in cluster %q", namespace, cluster.Name)
+	} else {
+		log.Debugf("Created namespace %q in cluster %q with label %s=%s",
+			namespace, cluster.Name, ramenE2EManagedLabel, ramenE2EManagedValue)
 	}
 
-	log.Debugf("Created namespace %q in cluster %q", namespace, cluster.Name)
-
 	return nil
 }
 
 func DeleteNamespace(ctx types.Context, cluster types.Cluster, namespace string) error {
 	log := ctx.Logger()
+	key := k8stypes.NamespacedName{Name: namespace}
+	objNs := &corev1.Namespace{}
 
-	ns := &corev1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: namespace,
-		},
+	err := cluster.Client.Get(ctx.Context(), key, objNs)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.Debugf("Namespace %q not found in cluster %q", namespace, cluster.Name)
+
+			return nil
+		}
+
+		return err
+	}
+
+	labels := objNs.GetLabels()
+	if labels == nil || labels[ramenE2EManagedLabel] != ramenE2EManagedValue {
+		log.Debugf("Namespace %q in cluster %q is not managed by ramen-e2e, skipping deletion",
+			namespace, cluster.Name)
+
+		return nil
 	}
 
-	err := cluster.Client.Delete(ctx.Context(), ns)
+	err = cluster.Client.Delete(ctx.Context(), objNs)
 	if err != nil {
 		if !k8serrors.IsNotFound(err) {
 			return err
 		}
 
 		log.Debugf("Namespace %q not found in cluster %q", namespace, cluster.Name)
-
-		return nil
 	}
 
 	return nil
 }
 
-// Problem: currently we must manually add an annotation to application’s namespace to make volsync work.
-// See this link https://volsync.readthedocs.io/en/stable/usage/permissionmodel.html#controlling-mover-permissions
-// Workaround: create ns in both drclusters and add annotation
-func CreateNamespaceAndAddAnnotation(ctx types.Context, namespace string) error {
-	env := ctx.Env()
-
-	if err := CreateNamespace(ctx, env.C1, namespace); err != nil {
+// CreateAppNamespaces creates a namespace on both drclusters with ramen-e2e label.
+// The label ensure safe deletion by allowing only ramen-e2e managed namespaces to be removed.
+// This prevents accidental deletion of namespaces not created by the e2e test framework.
+func CreateAppNamespaces(ctx types.Context, namespace string) error {
```

Comment:
If we can, lets split these changes, so we have 2 commits:
- First adding the infra for creating a namespace with our label and deleting only the namespaces we created
- Second using the infra to create, delete, and wait for namespaces in all deployers